notebooks/triplet_loss_cifar10.py

```diff
@@ -0,0 +1,236 @@
+"""
+This notebooks borrows from https://www.tensorflow.org/addons/tutorials/losses_triplet and is intended to compare tf.addons triplet loss
+implementation against this one. It is also aimed at benchmarking the impact of the distance function.
+"""
+from pathlib import Path
+from pprint import pprint
+
+import itertools
+import io
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow.keras.callbacks import TensorBoard
+from tensorflow.keras.layers import Conv2D, Dense, Dropout, GlobalMaxPooling2D, Input, Flatten, MaxPooling2D, Lambda
+from tensorflow.keras.models import Sequential
+
+from keras_fsl.layers import GramMatrix
+from keras_fsl.losses.gram_matrix_losses import binary_crossentropy, class_consistency_loss, triplet_loss
+from keras_fsl.metrics.gram_matrix_metrics import classification_accuracy
+from keras_fsl.utils.tensors import get_dummies
+
+
+#%% Build datasets
+train_dataset, val_dataset, test_dataset = tfds.load(
+    name="cifar10", split=["train[:90%]", "train[90%:]", "test"], as_supervised=True
+)
+train_dataset = train_dataset.shuffle(1024).batch(64, drop_remainder=True)
+val_dataset = val_dataset.shuffle(1024).batch(64, drop_remainder=True)
+test_dataset = test_dataset.batch(64, drop_remainder=True)
+
+train_labels = [batch[1].numpy().tolist() for batch in train_dataset]
+val_labels = [batch[1].numpy().tolist() for batch in val_dataset]
+test_labels = [batch[1].numpy().tolist() for batch in test_dataset]
```

Comment:
une remarque tout à fait optionnelle. Je trouve `[labels.numpy().tolist() for _, labels in train_dataset]` plus clair.